app/main/views/g_cloud.py

```diff
@@ -544,6 +544,65 @@ def end_search(framework_framework, project_id):
     )
 
 
+@direct_award.route(
+    '/<string:framework_framework>/projects/<int:project_id>/did-you-award-contract',
+    methods=['GET', 'POST']
+)
+def did_you_award_contract(framework_framework, project_id):
+    all_frameworks = data_api_client.find_frameworks().get('frameworks')
+    framework = framework_helpers.get_latest_live_framework(all_frameworks, framework_framework)
+
+    # Get the requested Direct Award Project.
+    project = data_api_client.get_direct_award_project(project_id=project_id)['project']
+
+    if not is_direct_award_project_accessible(project, current_user.id):
+        abort(404)
+
+    if project['lockedAt']:
+        abort(400)
+
+    form = DidYouAwardAContractForm()
+
+    if form.validate_on_submit():
+        if form.did_you_award_a_contract.data == 'still-assessing':
```

Comment:
I might consider a patch like the one below to avoid the magic strings, but I suspect we probably do it as you have in other places.

```diff
diff --git a/app/main/forms/direct_award_forms.py b/app/main/forms/direct_award_forms.py
index 454f7284..b7eb1885 100644
--- a/app/main/forms/direct_award_forms.py
+++ b/app/main/forms/direct_award_forms.py
@@ -25,12 +25,16 @@ class DMRadioField(RadioField):


 class DidYouAwardAContractForm(FlaskForm):
+    YES = 'yes'
+    NO = 'no'
+    STILL_ASSESSING = 'still-assessing'
+
     did_you_award_a_contract = DMRadioField(
         "Did you award a contract?",
         validators=[InputRequired(message="You need to answer this question.")],
         choices=[
-            ('yes', 'Yes'),
-            ('no', 'No'),
-            ('still-assessing', 'We are still assessing services')
+            (YES, 'Yes'),
+            (NO, 'No'),
+            (STILL_ASSESSING, 'We are still assessing services')
         ]
     )
diff --git a/app/main/views/g_cloud.py b/app/main/views/g_cloud.py
index b499fe89..1b35ffa9 100644
--- a/app/main/views/g_cloud.py
+++ b/app/main/views/g_cloud.py
@@ -564,15 +564,15 @@ def did_you_award_contract(framework_framework, project_id):
     form = DidYouAwardAContractForm()

     if form.validate_on_submit():
-        if form.did_you_award_a_contract.data == 'still-assessing':
+        if form.did_you_award_a_contract.data == form.STILL_ASSESSING:
             return redirect(url_for('.view_project',
                                     framework_framework=framework_framework,
                                     project_id=project_id))
-        elif form.did_you_award_a_contract.data == 'yes':
+        elif form.did_you_award_a_contract.data == form.YES:
             return redirect(url_for('.which_service_won_contract',
                                     framework_framework=framework_framework,
                                     project_id=project_id))
-        elif form.did_you_award_a_contract.data == 'no':
+        elif form.did_you_award_a_contract.data == form.NO:
             return redirect(url_for('.why_didnt_you_award_contract',
                                     framework_framework=framework_framework,
                                     project_id=project_id))
```